manifests/profile/puppetdb.pp

```diff
@@ -15,9 +16,9 @@
 # @param interval
 #   The frequency that telegraf will poll for metrics.  Defaults to '5s'
 define puppet_metrics_dashboard::profile::puppetdb (
-  String[2] $timeout                                          = $puppet_metrics_dashboard::params::http_response_timeout,
+  String[2] $timeout                                          = lookup('puppet_metrics_dashboard::http_response_timeout'),
   Variant[String,Tuple[String, Integer]] $puppetdb_host       = $facts['networking']['fqdn'],
-  Puppet_metrics_dashboard::Puppetdb_metric $puppetdb_metrics = $puppet_metrics_dashboard::params::puppetdb_metrics,
+  Puppet_metrics_dashboard::Puppetdb_metric $puppetdb_metrics = lookup('puppet_metrics_dashboard::puppetdb_metrics'),
```

Comment:
The puppetdb_metrics function should be rewritten to have no parameters and then just called directly as `puppet_metrics_dashboard::puppetdb_metrics()`

Reply:
The parameters here are a requirement for the hiera backend to work.  I'm still leaning towards using that, see my other comment below.

Reply:
I'm confused what the hiera backend is providing you that a regular puppet function doesn't?

Reply:
Functionally they do the same thing, however I'd never heard of using a function to resolve default parameter values until today, whereas module_data in hiera seems well used and documented.  

We needed a function here primarily to perform the logic around `puppetdb_metrics`, as they are dependent on PE version.  I'm not sure I'd want to add other functions in the future if we needed some other value, if we can provide them all in hiera here and now.

Reply:
The thing is you're just adding a layer of indirection to getting the data by having hiera call your function instead of calling the function directly.  It doesn't really make sense.

Reply:
it's also possible to workaround the situation without module data at all.  Something like

```
function puppet_metrics_dashboard::params(
  String $parameter_name
){
  include puppet_metrics_dashboard::params

  getvar("puppet_metrics_dashboard::params::${parameter_name}")
}
```

Then you'd just do `puppet_metrics_dashboard::params('puppetdb_metrics')` instead of `$puppet_metrics_dashboard::params::puppetdb_metrics`

Anyway, I think using module data is fine but for getting the puppetdb_metrics hash it should just be a straight function call not going through hiera.

Reply:
I haven't seen this pattern before. Are there other modules that leverage a custom hiera backend with the data contained within the function?

Reply:
@jarretlavallee I think the idea came from this: https://puppet.com/docs/puppet/5.5/hiera_migrate.html#concept-8159

Reply:
I am going to leave it to @suckatrash as to whethor or not to change, but I have zero objection to reworking this a little to be a normal function instead of the special Hiera backend. 

Doing that would mean the default value of `puppet_metrics_dashboard::puppetdb_metrics` in `init.pp` would no longer be pulled in via Hiera, but rather a function call a la  `puppet_metrics_dashboard::puppetdb_metrics = puppet_metrics_dashboard::some_function_name()`. 

If I understand correctly, the defined type's default value would change like so:

```diff
- Puppet_metrics_dashboard::Puppetdb_metric $puppetdb_metrics = lookup('puppet_metrics_dashboard::puppetdb_metrics'),
+ Puppet_metrics_dashboard::Puppetdb_metric $puppetdb_metrics = puppet_metrics_dashboard::some_function_name(),
```

Reply:
@genebean What @npwalker is saying makes sense to me now.  For some reason I thought he meant that we should throw out all the `module_data` changes, and those seem like an improvement to me.  I see the benefit of having the function bypass hiera and I do think it makes sense to implement it in that way.

`puppet_metrics_dashboard::puppetdb_metrics()` seems fine as a function name?

Reply:
Reworking now. Please do not merge yet... I am going to squash once I see tests pass.

Reply:
Resolving based on latest committed code.